jupyter-lib/api/src/main/kotlin/org/jetbrains/kotlinx/jupyter/api/libraries/JupyterIntegration.kt

```diff
@@ -69,9 +70,9 @@ abstract class JupyterIntegration : LibraryDefinitionProducer {
             fileAnnotations.add(handler)
         }
 
-        inline fun <reified T : Any> render(noinline renderer: (T) -> Any) {
+        inline fun <reified T : Any> render(noinline renderer: CodeCell.(T) -> Any) {
             val execution = ResultHandlerExecution { _, property ->
-                FieldValue(renderer(property.value as T), property.name)
+                FieldValue(renderer(notebook!!.currentCell!!, property.value as T), property.name)
```

Comment:
Probably better to provide error message instead of generic NPE if notebook is not defined.